FORMS.md

```diff
@@ -0,0 +1,243 @@
+
+### Content areas
+
+- Title: contains legend for FieldGroup
+- Descr: contains a description for a FieldGroup
+- Slot: contains Fields
+
+### Pseudo-Structure
+
+```html
+<FieldGroup>
+  <FieldGroup.Title>A Group of Fields</FieldGroup.Title>
+  <FieldGroup.Desc>This is a description of the FieldGroup.</FieldGroup.Desc>
+  [...]
+</FieldGroup>
+```
+
+### HTML
+
+```html
+<fieldset class="ods-field-group">
+  <legend class="ods-field-group--title">A Group of Fields</legend>
+  <p class="ods-field-group--desc">This is a description of the FieldGroup.</p>
+  [...]
+</fieldset>
+```
+
+## Form
+
+A container component that controls form behavior and provides layout slots.
+
+### Non-global attrs
+
+Form-specific HTML attributes we may need props for.
+
+#### Submission-related
+
+These values can be overridden by a `formaction` attribute on a `<button>`, `<input type="submit">`, or `<input type="image">` element.
+
+- `action`: The URL that processes the form submission.
+- `enctype`: If the value of the method attribute is post, enctype is the MIME type of the form submission. Possible values: `application/x-www-form-urlencoded` (default), `multipart/form-data` (use when form contains `<input type='file'>` elements), `text/plain` (debugging).
+- `method`: The HTTP method to submit the form with. Possible (case insensitive) values: `get`, `post`, `dialog`.
+- `novalidate`: This Boolean attribute indicates that the form shouldn't be validated when submitted.
+- `target`: Indicates where to display the response after submitting the form. The following values have special meanings: `_self` (default), `_blank`, `_parent`, `_top`.
+
+#### Likely unnec
+
+- `accept-charset`: Space-separated character encodings the server accepts. The default value means the same encoding as the page.
+- `autocomplete`: Indicates whether input elements can by default have their values automatically completed by the browser. autocomplete attributes on form elements override it on `<form>`. Possible values: `off`, `on`.
+  - WCAG requires that we specify this attribute on individual inputs that collect data about the user.
+- `name`: The name of the form. The value must not be the empty string, and must be unique among the `form` elements in the forms collection that it is in, if any.
+- `rel`: Creates a hyperlink or annotation depending on the value, see the rel attribute for details.
+
+### Content areas
+
+- Header: contains Title, Description
+- Main: contains FieldGroups, Fields
+- Error: contains an Infobox displaying Form-level errors
+- Footer: contains Form actions or action containers like ButtonGroup, also legalese
+
+### Pseudo-Structure
+
+```html
+<Form>
+  <Form.Header>
+    <Form.Title>Form Title</Form.Title>
+    <Form.Desc>This is a caption for the form.</Form.Desc>
+  </Form.Header>
+  <Form.Error>
```

Comment:
Why wrap the infobox in this Form.Error component? Does it provide any specific style? If so, what?

My assumption is that it just provides spacing. Could we get away with a grid layout and grid-gap for forms and just drop the infobox in here instead?

Reply:
As with the states above, I want to see what @joejezowski-okta comes back with. 

If we need to position Infobox in anyway way, I'd prefer to use a container because we're avoiding child selectors. If Infobox doesn't require anything specific, I'm 💯 on removing any HTML cruft and relying on Grid alone.

Reply:
Is the implication here that all form errors appear together in `Form.Error` separate from their fields?

I'd love to see a way to render error states for specific form field groups. Imagine a form where some group of fields is all required, but another group is not.

Reply:
There are two types of Errors that can be surfaced / are surfaced in the product today:

1. Field-level errors. These utilize the Field.Validation / Field.Error component and are a part of Field. 

<img width="396" alt="Screen Shot 2021-07-26 at 10 43 57 AM" src="https://user-images.githubusercontent.com/36284167/127034456-edd17a9e-ec59-40f5-b036-ba2ba15d6e0b.png">

2. The second is Form-level errors, which we don't have designs for in ODS but typically use a Callout/Infobox at the top of the form in our existing UIs. This same pattern could be used for Fieldset-level errors as @jeffbelser-okta suggests above. @joejezowski-okta can we do some exploration there?